ls1intum/Artemis
# `Programming exercises`: Fix feedback not shown when clicking on task in problem statement

### Checklist
#### General


#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
When clicking on a task in the problem statement the associated feedback was not shown as the component tried to access the exercise corresponding to the result but it was not available.

### Description
This PR updates the component to make sure the exercise is always available when the result details are shown.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Open the programming exercise in the online editor.
2. Click on some task in the problem statement that has feedback. A dialogue as shown in the screenshots below should come up where the feedback text is shown.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| `result-detail.component.ts` | ToDo | ToDo |


### Screenshots
The task in the problem statement shows up as clickable because it has some feedback text:
![Screenshot 2022-03-21 at 15-21-42 Code Editor](https://user-images.githubusercontent.com/64250573/159281652-86961864-386f-4615-a015-1d6460a238ca.png)
When clicking on it the feedback is not shown:
![Screenshot 2022-03-21 at 15-22-00 Code Editor](https://user-images.githubusercontent.com/64250573/159281655-db38dd90-5a19-4ff1-b22e-e7732c94c4e3.png)
Updated, the feedback is now shown as intended again when clicking on the task in the problem statement:
![Screenshot 2022-03-21 at 15-22-11 Code Editor](https://user-images.githubusercontent.com/64250573/159281657-66e9536a-f38a-4855-8a29-71fd19bacd80.png)
